xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -83,21 +99,20 @@ public static XdsServerBuilder forPort(int port, ServerCredentials serverCredent
   }
 
   @Override
-  public Server build() {
-    return buildServer(new XdsClientWrapperForServerSds(port));
+  public XdsServerWrapper build() {
+    checkState(isServerBuilt.compareAndSet(false, true), "Server already built!");
+    InternalNettyServerBuilder.eagAttributes(delegate, Attributes.newBuilder()
+            .set(ATTR_FILTER_CHAIN_SELECTOR_REF, filterChainSelectorRef)
+            .build());
+    return new XdsServerWrapper("0.0.0.0:" + port, delegate, timeService,
+            RETRY_DELAY_NANOS, xdsServingStatusListener, filterChainSelectorRef,
```

Comment:
`filterChainSelectorRef` really belongs to `XdsServerWrapper` so making it a member of `XdsServerBuilder` is not necessary. You can create `filterChainSelectorRef` either here and pass it to both  `eagAttributes` and `XdsServerWrapper` constrictor here or have `XdsServerWrapper` constructor create it and retrieve it from there to pass to `eagAttributes`.